clients/client-sts/src/defaultStsRoleAssumers.ts

```diff
@@ -84,17 +84,24 @@ export const getDefaultRoleAssumer = (
         logger: logger as any,
       });
     }
-    const { Credentials } = await stsClient.send(new AssumeRoleCommand(params));
+    const { Credentials, AssumedRoleUser } = await stsClient.send(new AssumeRoleCommand(params));
     if (!Credentials || !Credentials.AccessKeyId || !Credentials.SecretAccessKey) {
       throw new Error(`Invalid response from STS.assumeRole call with role ${params.RoleArn}`);
     }
+    let accountId;
+    try {
+      accountId = AssumedRoleUser!.Arn!.split(":")[4];
+    } catch (error) {
+      accountId = undefined;
+    }
```

Comment:
Us if instead of try-catch?
```suggestion
    let accountId;
    if (typeof AssumedRoleUser?.Arn === "string") {
      const arnComponets = AssumedRoleUser.Arn.split(":");
      if (arnComponets.length > 4 && arnComponets[4] !== "") {
        accountId = arnComponets[4];
      }
    }
```

Reply:
Since this is repeated on line 148, you can write a function

```js
const getAccountIdFromAssumedRoleUser = (assumedRoleUser?: AssumedRoleUser) => {
  if (typeof assumedRoleUser?.Arn === "string") {
    const arnComponets = AssumedRoleUser.Arn.split(":");
    if (arnComponets.length > 4 && arnComponets[4] !== "") {
      return arnComponets[4];
    }
  }
  return undefined;
};
```